nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ExecuteScriptMarkLogic.java

```diff
@@ -242,15 +241,14 @@ public final void onTrigger(final ProcessContext context, final ProcessSessionFa
             }
 
             originalFF = session.putAttribute(originalFF, MARKLOGIC_RESULTS_COUNT, Integer.toString(count));
-            session.transfer(originalFF, ORIGINAL);
 
             if (last != null && (count > 1 || !skipFirst)) {
                 FlowFile lastFF = session.create(originalFF);
                 resultToFlowFile(session, last, lastFF, resultsDest);
                 session.transfer(lastFF, LAST_RESULT);
             }
 
-            session.commitAsync();
+            transferAndCommit(session, originalFF, ORIGINAL);
```

Comment:
Cutting down on the number of occurrences of `commitAsync` - now all processors are using our `transferAndCommit` convenience method.